user/super/com/google/gwt/emul/java/util/Optional.java

```diff
@@ -104,6 +129,10 @@ public T orElseGet(Supplier<? extends T> other) {
     return isPresent() ? ref : other.get();
   }
 
+  public T orElseThrow() {
```

Comment:
This is Java 10 API. It is the only Optional[Double/Int/Long] Method added in Java 10. 

Do we want dedicated Java 10 tests? If yes, we should remove this method in this PR.

Reply:
My inclination was to split by java version and by "where in the emulation" the feature is added, otherwise the full set of patches is huge, and reviewers will only get through the first hundred lines and stop reading as closely, so yes, I'll split this off too.